File: src/main/java/com/anyneeds/auth/service/AuthService.java
package com.anyneeds.auth.service;

import com.anyneeds.auth.dto.AuthResponse;
import com.anyneeds.common.JwtUtil;
import com.anyneeds.user.entity.User;
import com.anyneeds.user.service.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class AuthService {

    private final OtpService otpService;
    private final UserService userService;
    private final JwtUtil jwtUtil;

    public void sendOtp(String phoneNumber) {
        otpService.sendOtp(phoneNumber);
    }

    public AuthResponse verifyOtp(String phoneNumber, String otpCode) {
        boolean valid = otpService.verifyOtp(phoneNumber, otpCode);
        if (!valid) throw new RuntimeException("Invalid or expired OTP");

        User user = userService.findOrCreateUser(phoneNumber);
        String token = jwtUtil.generateToken(user.getId(), user.getPhoneNumber());

        return AuthResponse.builder()
            .token(token)
            .user(com.anyneeds.user.dto.UserProfileDto.from(user))
            .build();
    }
}

File: anyneeds-ui/src/services/authService.js
import api from './api';

export const sendOtp = (phoneNumber) =>
  api.post('/api/auth/send-otp', { phoneNumber });

export const verifyOtp = (phoneNumber, otpCode) =>
  api.post('/api/auth/verify-otp', { phoneNumber, otpCode });

export const getProfile = () => api.get('/api/users/me');

export const updateProfile = (data) => api.put('/api/users/me', data);
